store/json.go

```diff
@@ -153,21 +147,20 @@ func (kvs *jsonFileStore) Lock(block bool) error {
 			break
 		}
 
-		if !block || i == lockMaxRetries {
-			return ErrStoreLocked
+		if !block {
+			return ErrNonBlockingLockIsAlreadyLocked
 		}
 
-		time.Sleep(time.Duration(lockRetryDelayInMilliseconds) * time.Millisecond)
-	}
+		if i == lockMaxRetries {
+			return ErrTimeoutLockingStore
+		}
 
-	// Write the process ID for easy identification.
-	_, err = lockFile.WriteString(strconv.Itoa(os.Getpid()))
-	if err != nil {
-		return err
+		time.Sleep(lockRetryDelay)
 	}
+	defer lockFile.Close()
```

Comment:
I'm not sure how defer works in a loop . Just clarifying if open called multiple times and is close getting called for each open?

Reply:
sorry! I missed it